### ISSUE TITLE
Too difficult to follow

### ISSUE THREAD

**Issue by er-shrey** [exercism/exercism#6333]
1. Instructions in the editor are not matching or are irrelevant.
2. Results of the iterations goes away when I open editor
3. Application is too tough for a general user
4. There is no feedback option
5. People will come here to learn the programming language or any topic... But most of the time is taken by your application, seems like we are here to learn your application only.
6. There is no way that I can move to next exercise from one exercise. Why so complexity?

**Comment by iHiD** [comment]
Thanks for the feedback. Could you give some examples of these points please. For example, where do the instructions now match, where are you not finding the option to move to a different exercise, where are you wanting feedback where you can't get it? Please provide tangible examples and screenshots. Thank you.

**Comment by SaschaMann** [comment]
> There is no way that I can move to next exercise from one exercise. Why so complexity?

I've had the same issue before. When I complete a (learning) exercise, it prompts me to either go back to the exercise (I just completed it, why would I do this?) or to go to the concepts. Usually I either want to end the session or start the next exercise. Figuring out which exercise is meant to be next isn't trivial and requires some knowledge of the UI and clicking around. A "do the next exercise" button would help here.

**Comment by iHiD** [comment]
> Figuring out which exercise is meant to be next isn't trivial and requires some knowledge of the UI and clicking around. A "do the next exercise" button would help here.

This is because there's not a concept of a "Next lesson". Some people go and choose to go broad (ie another Learning Exercise) and some people choose to go deeper (ie practice with Practice Exercises). In either scenario there's normally multiple choices as to what you can do. The best way to navigate this decision (in either path) is to use the Concept Tree.

Maybe we should provide a "next lesson", but after many hours of discussion during the build we couldn't come to any consensus about what that actually means, and definitely different people would have very different expectations about what it would mean.

**Comment by BethanyG** [comment]
This flow has also confusing for me. 

Would it make sense to offer 2-3 "CTA" options along the lines of:

1.  `[Next Concept]` - the next concept exercise (if there are learning exercises) 
2.  `[Practice]` - the next (and easiest by difficulty) practice exercise that practices the concept exercise that has just been completed.
3.  `[Another Exercise]` - the next (and easiest by difficulty) practice exercise in the `config.json` order for the track, if the track has no learning exercises.  Or this could take them to  `https://exercism.org/tracks/<track>/exercises?status=available`

I think concepts page does this, more or less?  So (now thinking out loud) maybe its two options:

1.  "See what's next with `<concept>`  -->  goes to `https://exercism.org/tracks/<track>/concepts/<concept>`
2.  "Select your next concept to learn "  -->  goes to syllabus page
3.  (_for tracks that don't have learning exercises_) "Select your next exercise" --> goes to `https://exercism.org/tracks/<track>/exercises?status=available`.

**Comment by glaxxie** [comment]
Hi, I just want to contribute some of my thoughts regarding the flow of one of the track.
My main language track is Python and I've been using the site around half the year to start improving my skill. I've done about 
2/3 of all the exercises (total 131 at the point of writing this)

* Learning exercises: these are great small introduction to a new concept, but it is exactly what it is - an introduction. So students should be encouraged to learn more about these concepts from the web, doing small, quick exercises to get a grasp of the idea and be more familiar with it before tackling the problem. Basically having a good fundamental understanding before starting the exercises due to reasons below.

* The syllabus: the syllabus page show a broad overview of the fundamental concepts of the programming language : basics -> numbers , bools -> conditions etc. However the exercises inside the sectioned concepts feel like they are in a somewhat arbitrary order, I guess one reason is you have to unlock other concepts and exercises to go back and unlocking some more from a previous one.  But even then it still feel a bit disorientated. 
  - Exercises ordering : in the Bools section, the first exercise is the learning exercise to introduce user to the concept, then 2nd is Luhn a  medium problem, then next is Triangle an easy problem, follow by 2 medium problems and suddenly a hard one with Rest Api (which I assume user gonna need to know more than just bools, then it goes back to 2 easy and 1 medium.
 
* Exercises unlocking : so due to the syllabus page does not exactly behaving in a linear fashion, I mostly refer to the Exercises page as my place to start, and start working from the lowest difficulties. Which I then encounter a funny situation : for a while I got locked out of the exercise "Ellen's Alien Game", which is a learning exercise to introduce user into OOP and class because I missed out some other exercises to unlock it. Yet there are other exercises that explicitly deal with classes and OOP already available to do.
I also mistakenly did the "Nth Prime" exercise before I realize there is a related one in easy difficulty as "Sieve ". This is totally my fault for not looking closely, but I wish the "Nth Prime" was locked first and then got unlock after you have to deal with "Sieve", because some of the later problems are not simply just basic concepts anymore but you have to think algorithmically and know some of them.

* The Difficulty: Probably not much needed to say here since I'm aware this is an open issue and I already saw the aggregated list. But yeah, not all easy problems are created equally. 

*Grading : sometimes I got the code to run and behave exactly as wanted, but the grading demand a different kind of input or output. However I totally understand why since it is machine graded, so it is not a real issue for me.

Those are some issues I thought of while using the site, overall it has been a great experience. I've been able to learn more about wide variety of programming concepts, algorithms, history, and some actual real life application from these exercises. I would continue using the website and encourage others to try it as well.

tl;dr : might be a bit overwhelming for totally newbie, syllabus layout is not very useful, unlocking exercises feels unnatural sometimes, and difficulties need a revamp. still a great website for practicing, improving and learning more

ps: i don't know if anybody ever ask but a dark mode would be amazing

**Comment by BethanyG** [comment]
@glaxxie  -- as a Python track maintainer, your comments are super-helpful, so thank you.  We can't address all of them, but the points about ordering of practice exercise difficulty within the concept categories we can certainly work on, as well as overall difficulty ratings and OOP vs non-OOP.  --or at least I _think_ we can work on improving those via our track configuration.  Would you mind if I copied your comment over to the Python track and made an issue?

**Comment by glaxxie** [comment]
@BethanyG Please feel free to copy/edit it in whatever way that you need. It's the least I could do to help contributing for the moment.

**Comment by SaschaMann** [comment]
> This is because there's not a concept of a "Next lesson". Some people go and choose to go broad (ie another Learning Exercise) and some people choose to go deeper (ie practice with Practice Exercises). In either scenario there's normally multiple choices as to what you can do. The best way to navigate this decision (in either path) is to use the Concept Tree.

To understand the reasoning behind this and to figure out what one wants to do after solving the exercise, one needs to understand the following things:

- The difference between learning and practice exercises
- The relationship between concepts and learning exercises
- The relationship between concepts and practice exercises
- How to read the UI in the concept tree to identify the exercises

That's *a lot* of Exercism-specific knowledge required to figure out what exercise to do next.

I don't want to derail this into a discussion about what the next lesson should be[^1], I wanted to outline the reason for why it is really confusing to move on to the next exercise. Bethany and I both know Exercism and the principles behind it well, yet we both seemed to have struggled with this flow before. For someone new to the site it must be even more confusing.

[^1]: though I have some thoughts, so feel free to start it elsewhere :P

**Comment by SleeplessByte** [comment]
(@glaxxie [Dark mode will be coming in the future](https://github.com/exercism/exercism/issues?q=dark+mode))

@iHiD One of the reasons I've been pretty hesitant to add the practice exercises "practices" key to JavaScript is that I found Ruby and Python's tree very difficult to traverse, and Tom who was practicing Ruby also got stuck right after the second learning exercise because He thought he needed to finish "all of them" before moving on to the "next subject".

Going through "just" the syllabus by itself seems, for JavaScript, Ruby, and Python, as well as Elixir and C#, by my own account, a decent path, and whilst there are branches, having only the choice from the next _concept_ exercise actually provides a pretty nice path forward.

I know it's not what we want or what's optimal, but if we make any short term decision, that may be it?

Fully agree with @SaschaMann on the offloading point.

**Comment by junedev** [comment]
It was on my todo list to open an issue about this topic as well. I had some screenshots saved for that. I will leave them here as they might help others to understand what we are talking about.

Here is the concept exercise completed modal:

![image](https://user-images.githubusercontent.com/12543047/165622174-4343adf0-9492-4e52-960f-6e612de8c63a.png)

I agree with the people that are not happy with the choices here. Maybe if the exercise list on there would be along the lines of "Select your next exercise" and there would be the usual hover with the blurb and the items are clickable, that would already help.

Because otherwise there is the syllabus or this page which is the one that bugs me the most: 

![image](https://user-images.githubusercontent.com/12543047/165622075-30a7e012-1866-4ce9-ab1d-997cd93646c6.png)

My brain always overloaded when looking at this page (not just after completing an exercise but also in other states in can have). There are tabs and banners and concept and exercise lists and various primary and secondary buttons, a sidebar, a button with dropdown above the sidebar that often seems to be the same as the button in the sidebar etc.
It took me a lot of times of seeing the page before launch to grasp what's going on. I am happy to provide some create improvement ideas in case we decide to work on this page at some point.

**Comment by iHiD** [comment]
Thanks all. I think there are three different (general) points that are worth discussing individually here:
1. Adding a "Next exercise button" - it feels like "Next Learning Exercise" is the route to go?
2. Making it clearer what is expected of you as a user in terms of breadth/depth (e.g. DJ's example of Tom's expectations)
3. The complexity of the Exercise Page.

If everyone agrees those are the three pertinent things to discuss here, can you give this a :+1: please and then we can split those into different discussions. If I've missed something you feel is another discussion, can you add that below please and we'll create seperate discussions for those too.

**Comment by pokrakam** [comment]
If I may add a short comment, I also agree the `Return to Exercise` button is not logical at the point the user has just indicated that they're done. As someone who joined Exercism as both both track author and student (and developer of many years), I found the distinction between concept and practice confusing and it took me some time to get the hang of it. 
See also https://github.com/exercism/abap/issues/98

The other point not mentioned is that there are many conceptless practice exercises, so I don't think the "Next Exercise" has a simple answer. Maybe a random/suggested exercise, but I fear someone following a syllabus may finish a simple concept and be given an easy exercise that nevertheless needs advanced knowledge. My preference would be to have a single "Return to Overview" button to go back to the overview page with all the little exercise icons, seeing the grid with tick marks give a sense of accomplishment.

**Comment by asarkar** [comment]
My 2c. I appreciate the work that goes behind the platform, but I too find the UI/workflow clunky. I've created a ticket to address some of the concerns. https://github.com/exercism/exercism/issues/6376

**Comment by ToxicSmurf** [comment]
Exercism is my favorite FOSS learning platform bar none. The only page I find myself regularly having to strain my brain on is the dashboard. I know nothing about UI/UX, but it feels like the total amount of elements/information on that page overheats my brain for just a second or two.

**Comment by mshprygov** [comment]
Just add **one big green "DONE!" button** on the exercise completion, that brings you to the _current track page_.

That must be _that_ simple!
